blueprints/to_soa.md

```diff
@@ -0,0 +1,159 @@
+          "hostname": "to.backend.net",
+          "port": 8445
+        }
+      ],
+      // A boolean specifying whether or not TO should verify the backend server's certificate chain and host name. This is not recommended for production use.
+      "insecure": false,
+      // Permissions required to access the route
+      "permissions": [
+        "FOO:READ"
+      ],
+      // ID for the route
+      "routeId": 123456,
+      // Extra options to configure how TO forwards these requests
+      "opts": {
+        // Currently, TO just supports a round robin algorithm to forward the requests.
+        "alg": "roundrobin"
+      }
+    }
+  ]
+}
+```
+
+#### REST API Impact
+n/a
+
+#### Client Impact
+Clients will still send requests and receive responses (for the newly configured endpoints) from TO, so nothing will change from the clients' perspective.
+
+#### Data Model / Database Impact
+There is no impact on the existing TO database structure/ schema. The backend services are free to have their own databases and 
+serve data from there. As before, only TO will be allowed to access the Traffic Ops Database (TODB).
+
+### t3c Impact
+n/a
+
+### Traffic Monitor Impact
+n/a
+
+### Traffic Router Impact
+n/a
+
+### Traffic Stats Impact
+n/a
+
+### Traffic Vault Impact
+n/a
+
+### Documentation Impact
+Documentation will have to be added for the new configuration option for TO.
+
+### Testing Impact
+The backend services will be responsible for writing their own test suites.
+
+### Performance Impact
+We don't foresee any performance impact on TO as a result of this architecture change.
+
+### Security Impact
+SSL certificates will be required for HTTPS protocol to be used between TO and the services. These can be implemented however is preferred by the 
+backend service. The overhead will include certificate generation and signing, certificate deployment to services' servers, certificate management 
+and renewal. Client certificates are an option for the future but will require further development.
+
+### Upgrade Impact
+If a change has to go in to one of the backend services, a redeploy/ restart of TO will no longer be necessary. The backend service will, however, 
+need to be re deployed. After that, if there is no change to the backend configuration file, no action will be required on the TO side. Else, as explained 
+earlier, a SIGHUP signal will need to be sent to TO to reload the new configuration values.
+
+### Operations Impact
+The added backend services will add some overhead to operations. However, this architecture will also reduce the load on operations from the TO side, since 
+every code change now won't force redeploying TO as a whole. 
+
+### Developer Impact
+If a new endpoint (or refactor of an existing one) is deemed as a fit candidate for a backend service, the developer will have to undertake the tasks of writing
+this new service (which can be a simple server serving these endpoints), and the tests associated with it. 
```

Comment:
So the plan is that anything that has the ability to be in the TO core, i.e., the functionality of the new API route can affect more than one backend service, goes in the TO core. If the feature has is an "add-on", it gets added as a backend service. This is mainly for organizations' internal business logic, but there is nothing stopping the OSS community from using thie feature as well.

Reply:
Seems like #6810 would want to make use of it, at least.

Is this meant to replace the existing plugin system, or would those still have a purpose moving forward?

Reply:
Plugins, though, have access to the Traffic Ops database, so they currently can't all be implemented as back-end services as described in this blueprint because:

> As before, only TO will be allowed to access the Traffic Ops Database

So if the intention is to deprecate the existing plugin system, then should services have access to the TODB?

Reply:
You're right. The idea is not to deprecate the plugin system. We'll have the plugins as they are, going forward. This architecture will be used by backend services that have their own databases and don't need to access the TODB.

Reply:
Okay, that's probably fine